core/src/main/scala/kafka/coordinator/transaction/TransactionLog.scala

```diff
@@ -266,6 +266,29 @@ object TransactionLog {
       }
     }
   }
+
+  /**
+   * Exposed for printing records using [[kafka.tools.DumpLogSegments]]
+   */
+  def formatRecordKeyAndValue(record: Record): (Option[String], Option[String]) = {
+    val txnKey = TransactionLog.readTxnRecordKey(record.key)
+    val keyString = s"transaction_metadata::transactionalId=${txnKey.transactionalId}"
+
+    val txnMetadata = TransactionLog.readTxnRecordValue(txnKey.transactionalId, record.value)
```

Comment:
Should we change `readTxnRecordValue` to return an Option? Glancing at the code, it looks like another possible NPE in TransactionStateManager#loadTransactions

Reply:
Yeah, good suggestion. Pushed a change.